client2/app/components/AddAssignmentForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactNode, useState } from "react";
+
+type AssignmentFormInfo = {
+  title: string;
+  score: number;
+  description: string;
+  category: string;
+};
+
+export const AddAssignmentForm = ({
+  accountId,
+}: {
+  accountId: string;
+}): ReactNode => {
+  const [assignment, setAssignment] = useState<AssignmentFormInfo>({
+    title: "",
+    score: 0,
+    description: "",
+    category: "",
+  });
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { name, value } = event.target;
+    setAssignment({
+      ...assignment,
+      [name]: name === "score" ? Number(value) : value,
+    });
+  };
+
+  const submitAssignment = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const payload = JSON.stringify({
+      accountId,
+      ...assignment,
+    });
+    const response = await fetch("http://localhost:8080/api/v2/assignments", {
```

Comment:
Might want to extract the routes to a different class to have it more dynamic, like a config.ts or use the native api setup in next (can read up on that part on next docs)